src/IBadgeQuery.sol

```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+/**
+ * @title IBadgeQuery
+ * @dev Interface for querying badge ownership.
+ */
+interface IBadgeQuery {
+    /**
+     * @notice Checks if an address holds a specific badge.
+     * @param account The address to query.
+     * @param badgeId The ID of the badge to check.
+     * @return hasBadge A boolean indicating whether the address holds the badge.
+     */
+    function hasBadge(address account, uint256 badgeId) external view returns (bool hasBadge);
```

Comment:
I think it's better to have badges as bytes32 as it leaves more flexibility, although we have integers on internal db they can still be passed as 32bytes

Reply:
I've done that in ThriveWorkerUnit but then thought maybe its easier this way.